algorithms/unit_tests/TestSort.hpp

```diff
@@ -402,6 +403,339 @@ void test_sort_integer_overflow() {
+      SizeType begin = offsets(i);
+      SizeType end   = offsets(i + 1);
+      if (sortDescending)
+        Kokkos::Experimental::sort_thread(
+            t, Kokkos::subview(keys, Kokkos::make_pair(begin, end)),
+            GreaterThan<KeyType>());
+      else
+        Kokkos::Experimental::sort_thread(
+            t, Kokkos::subview(keys, Kokkos::make_pair(begin, end)));
+    }
+  }
+  KeyViewType keys;
+  OffsetViewType offsets;
+  bool sortDescending;
+};
+
+// Functor to test sort_by_key_thread
+template <typename ExecSpace, typename KeyViewType, typename ValueViewType,
+          typename OffsetViewType>
+struct ThreadSortByKeyFunctor {
+  using TeamMem  = typename Kokkos::TeamPolicy<ExecSpace>::member_type;
+  using SizeType = typename KeyViewType::size_type;
+  using KeyType  = typename KeyViewType::non_const_value_type;
+  ThreadSortByKeyFunctor(const KeyViewType& keys_, const ValueViewType& values_,
+                         const OffsetViewType& offsets_, bool sortDescending_)
+      : keys(keys_),
+        values(values_),
+        offsets(offsets_),
+        sortDescending(sortDescending_) {}
+  KOKKOS_INLINE_FUNCTION void operator()(const TeamMem& t) const {
+    int i = t.league_rank() * t.team_size() + t.team_rank();
+    // Number of arrays to sort doesn't have to be divisible by team size, so
+    // some threads may be idle.
+    if (i < offsets.extent_int(0) - 1) {
+      SizeType begin = offsets(i);
+      SizeType end   = offsets(i + 1);
+      if (sortDescending) {
+        Kokkos::Experimental::sort_by_key_thread(
+            t, Kokkos::subview(keys, Kokkos::make_pair(begin, end)),
+            Kokkos::subview(values, Kokkos::make_pair(begin, end)),
+            GreaterThan<KeyType>());
+      } else {
+        Kokkos::Experimental::sort_by_key_thread(
+            t, Kokkos::subview(keys, Kokkos::make_pair(begin, end)),
+            Kokkos::subview(values, Kokkos::make_pair(begin, end)));
+      }
+    }
+  }
+  KeyViewType keys;
+  ValueViewType values;
+  OffsetViewType offsets;
+  bool sortDescending;
+};
+
+// Randomly generate n arrays of up to k elements each, packed into a single 1D
+// view. Like a row in a StaticCrsGraph, array i begins at offsets(i)
+// (inclusive) and ends at offsets(i+1) (exclusive)
+template <typename KeyType, typename KeyViewType, typename OffsetViewType>
+void randomPackedArrays(unsigned n, unsigned k, KeyViewType& keys,
+                        OffsetViewType& offsets, KeyType minKey,
+                        KeyType maxKey) {
+  offsets          = OffsetViewType("Offsets", n + 1);
+  auto offsetsHost = Kokkos::create_mirror_view(Kokkos::HostSpace(), offsets);
+  Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace> gHost(1931);
+  Kokkos::fill_random(offsetsHost, gHost, k);
+  Kokkos::DefaultHostExecutionSpace().fence();
+  offsetsHost(0) = 0;
+  // Prefix-sum
+  for (unsigned i = 1; i <= n; i++) {
+    offsetsHost(i) += offsetsHost(i - 1);
+  }
```

Comment:
Now that we require C++17 prefer `std::exclusive_scan`

Reply:
I would like to use this, but it seems like the standard tends to be missing some things in the 17 standard. For example, this:
```
#include <numeric>
#include <iostream>
#include <vector>

int main()
{
  std::vector<int> myVec = {1, 2, 3, 4};
  std::exclusive_scan(myVec.begin(), myVec.end(), myVec.begin(), 0);
  for(auto i : myVec)
    std::cout << i << ' ';
  std::cout << '\n';
  return 0;
}
```
doesn't compile for me because all the overloads expect an ``ExecutionPolicy&&`` as the first argument. If I try to ``#include <execution>`` to provide one, I get ``tbb/blocked_range.h: no such file or directory``.
I have GCC 9.1.0, so it's not a super old version (and the new Kokkos minimum is 8.2). Even on kokkos-dev2 with GCC 10.1.0, you can't include ``<execution>`` because the TBB is apparently too old...

Reply:
Looking at https://en.cppreference.com/w/cpp/compiler_support/17#C.2B.2B17_library_features it looks like you are right